crates/store/re_chunk/src/chunk.rs

```diff
@@ -218,6 +218,16 @@ impl Chunk {
             .collect()
     }
 
+    #[inline]
+    pub fn num_events(&self) -> usize {
+        self.components.values().map(|array| array.len()).sum()
+    }
+
+    #[inline]
+    pub fn num_events_for_component(&self, component_name: ComponentName) -> Option<usize> {
+        self.components.get(&component_name).map(|v| v.len())
+    }
+
```

Comment:
3 issues here:
* `num_events` is misleading, as a caller you'd think this is an alias for `Self::num_rows`.
* You cannot rely on the length of the underlying `ListArray` since component columns are sparse.
* Precise naming of arrays is super important in `re_chunk` because everything is some kind of arrow array in here; it quickly gets unintelligible.

```suggestion
    /// The cumulative number of events in this chunk.
    ///
    /// I.e. how many _component batches_ ("cells") were logged in total?
    //
    // TODO(cmc): This needs to be stored in chunk metadata and transported across IPC.
    #[inline]
    pub fn num_events_cumulative(&self) -> usize {
        // Reminder: component columns are sparse, we must take a look at the validity bitmaps.
        self.components
            .values()
            .map(|list_array| {
                list_array.validity().map_or_else(
                    || list_array.len(),
                    |validity| validity.len() - validity.unset_bits(),
                )
            })
            .sum()
    }

    /// The number of events in this chunk for the specified component.
    ///
    /// I.e. how many _component batches_ ("cells") were logged in total for this component?
    //
    // TODO(cmc): This needs to be stored in chunk metadata and transported across IPC.
    #[inline]
    pub fn num_events_for_component(&self, component_name: ComponentName) -> Option<usize> {
        // Reminder: component columns are sparse, we must check validity bitmap.
        self.components.get(&component_name).map(|list_array| {
            list_array.validity().map_or_else(
                || list_array.len(),
                |validity| validity.len() - validity.unset_bits(),
            )
        })
    }